[SwiftUI] Update the set of supported view modifiers on WebView (part 2)

rr-codes opened the pull request:
#### 9bc17785e12e3a4073aa04862f811df982d66802
<pre>
[SwiftUI] Update the set of supported view modifiers on WebView (part 2)
<a href="https://bugs.webkit.org/show_bug.cgi?id=290343">https://bugs.webkit.org/show_bug.cgi?id=290343</a>
<a href="https://rdar.apple.com/144668661">rdar://144668661</a>

Reviewed by Aditya Keerthi.

Refine the signature of the WebView context menu view modifier.

* Source/WebKit/_WebKit_SwiftUI/API/View+WebViewModifiers.swift:
(View.webViewContextMenu(@ViewBuilder menu: @MainActor @escaping (WebView.ActivatedElementInfo) -&gt; some View:)):

- Use `some View` instead of a generic, as opaque types are preferred when possible.
- Annotate the closure as `@MainActor`, since it will be invoked on the main actor.
- Use the new `WebView.ActivatedElementInfo` type instead of the prior `WebPage.ElementInfo` type
- Mark the availability as macOS only.

* Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:

Add the new `WebView.ActivatedElementInfo` type, which is effectively the same as `WebPage.ElementInfo` was, just renamed.

* Source/WebKit/UIProcess/API/Cocoa/WKContextMenuElementInfoAdapter.swift: Renamed from Source/WebKit/UIProcess/API/Swift/WebPage+ElementInfo.swift.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Since the element info type is now part of WebView and not WebPage, it is only accessible by the cross-import overlay.
As a result, an adapter type is needed to be able to communicate back to WebPage from WebView.

* Source/WebKit/UIProcess/API/Swift/WebPage.swift:
(setMenuBuilder(_:)):
* Source/WebKit/UIProcess/Cocoa/WKUIDelegateAdapter.swift:
(WKUIDelegateAdapter.menuBuilder):
(WKUIDelegateAdapter._webView(_:getContextMenuFromProposedMenu:forElement:userInfo:)):
* Source/WebKit/_WebKit_SwiftUI/Implementation/ViewModifierContexts.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/WebViewRepresentable.swift:
(WebViewRepresentable.updatePlatformView(_:context:)):

Update the relevant methods and callsites to use the new types.

Canonical link: <a href="https://commits.webkit.org/292708@main">https://commits.webkit.org/292708@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/05f701accfb71dc4b60f988c3520c83a591750b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96717 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16331 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6477 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101790 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47237 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98762 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16627 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24770 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73707 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30927 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99720 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12516 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87475 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54042 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12273 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5278 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46565 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82375 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5365 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103813 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23785 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17345 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82756 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/24035 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83532 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82142 "Found 3 new API test failures: /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/print-errors, /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/print, /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/close-after-print (failure)") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26794 "Build is in progress. Recent messages:OS: Sequoia (15.2), Xcode: 16.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; Passed layout tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4328 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17261 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15612 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23747 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/28902 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23406 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26886 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25147 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
```diff
@@ -138,4 +138,18 @@ extension WebView {
     }
+
+    /// Contains information about an element the user activated in a webpage, and which you use to configure a context menu for that element.
+    /// For links, the information contains the URL that the link represents.
```
```suggestion
    /// Contains information about an element the user activated in a webpage, which may be used to configure a context menu for that element.
    /// For links, the information contains the URL that is linked to.
```

rr-codes committed:
[SwiftUI] Update the set of supported view modifiers on WebView (part 2)
https://bugs.webkit.org/show_bug.cgi?id=290343
rdar://144668661

Reviewed by Aditya Keerthi.

Refine the signature of the WebView context menu view modifier.

* Source/WebKit/_WebKit_SwiftUI/API/View+WebViewModifiers.swift:
(View.webViewContextMenu(@ViewBuilder menu: @MainActor @escaping (WebView.ActivatedElementInfo) -> some View:)):

- Use `some View` instead of a generic, as opaque types are preferred when possible.
- Annotate the closure as `@MainActor`, since it will be invoked on the main actor.
- Use the new `WebView.ActivatedElementInfo` type instead of the prior `WebPage.ElementInfo` type
- Mark the availability as macOS only.

* Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:

Add the new `WebView.ActivatedElementInfo` type, which is effectively the same as `WebPage.ElementInfo` was, just renamed.

* Source/WebKit/UIProcess/API/Cocoa/WKContextMenuElementInfoAdapter.swift: Renamed from Source/WebKit/UIProcess/API/Swift/WebPage+ElementInfo.swift.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Since the element info type is now part of WebView and not WebPage, it is only accessible by the cross-import overlay.
As a result, an adapter type is needed to be able to communicate back to WebPage from WebView.

* Source/WebKit/UIProcess/API/Swift/WebPage.swift:
(setMenuBuilder(_:)):
* Source/WebKit/UIProcess/Cocoa/WKUIDelegateAdapter.swift:
(WKUIDelegateAdapter.menuBuilder):
(WKUIDelegateAdapter._webView(_:getContextMenuFromProposedMenu:forElement:userInfo:)):
* Source/WebKit/_WebKit_SwiftUI/Implementation/ViewModifierContexts.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/WebViewRepresentable.swift:
(WebViewRepresentable.updatePlatformView(_:context:)):

Update the relevant methods and callsites to use the new types.

Canonical link: https://commits.webkit.org/292708@main
